src/test/java/com/cloudant/tests/SslAuthenticationTest.java

```diff
@@ -0,0 +1,286 @@
+package com.cloudant.tests;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import com.cloudant.client.api.CloudantClient;
+import com.cloudant.client.api.model.ConnectOptions;
+import com.cloudant.test.main.RequiresCloudantService;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.lightcouch.CouchDbException;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLServerSocket;
+import javax.net.ssl.SSLServerSocketFactory;
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.SSLHandshakeException;
+import java.io.BufferedWriter;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.net.SocketException;
+import java.security.KeyStore;
+
+
+public class SslAuthenticationTest {
+
+    private static final Log log = LogFactory.getLog(SslAuthenticationTest.class);
+    private static CloudantClient dbClient;
+    private static HttpsServer server;
+
+    private static String KEYSTORE_FILE = "src/test/resources/SslAuthenticationTest.keystore";
+    private static String KEYSTORE_PASSWORD = "password";
+    private static String KEY_PASSWORD = "password";
+    private static int LOCAL_SERVER_PORT = 65535;
+    private static String LOCAL_SERVER_URL = "https://127.0.0.1:" + LOCAL_SERVER_PORT;
+
+    private static String DUMMY_USERNAME = "username";
+    private static String DUMMY_PASSWORD = "password";
+
+    private static boolean localServerReady;
+    private static final Object lock = new Object();
+
+    /**
+     * A very simple HTTPS Server that runs on the localhost and has a simple certificate that won't
+     * work for hostname verification.
+     * <p>
```

Comment:
I believe `<p>` is correct for Javadoc. It's one of those weird things that's inconsistent with normal HTML/XML formatting - see the 7th bullet under *Format of a Doc Comment* http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#format